app/util/im2db_utils.py

```diff
@@ -0,0 +1,165 @@
+        #        "study_date": study_date,
+        #        "study_time": study_time,
+        #        "station_name": station_name,
+        #        "accession_number": accession_number
+        #        })
+
+        # Save the image data into the corresponding tables:
+        # 0. Device:
+        try:
+            device_id = Device.query.where(Device.institution == institution
+                ).where(Device.manufacturer == manufacturer
+                ).where(Device.device_model == model
+                ).where(Device.station_name == station_name).first().id
+        except:
+            new_device = Device(
+                institution=institution, manufacturer=manufacturer,
+                device_model=model, station_name=station_name)
+            new_device.save()
+            db.session.add(new_device)
+            device_id = new_device.id
+        #TODO remove in production
+        print("device id:", device_id)
+        
+        # 1. Study:
+        try:
+            study_id = Study.query.filter_by(uid=study_uid).first().id
+        except:
+            new_study = Study(uid=study_uid,
+                              description=study_description,
+                              study_date=study_date)
+            new_study.save()
+            db.session.add(new_study)
+            study_id = new_study.id
+        #TODO remove in production
+        print("study id:", study_id)
+        
+        # 2. Series:
+        try:
+            series_id = Series.query.filter_by(uid=series_uid).first().id
+        except:
+            series_datetime = datetime.strptime("-".join(
+                                [dcm.SeriesDate, dcm.SeriesTime.split('.')[0]]
+                                    ), '%Y%m%d-%H%M%S')
+
+            new_series = Series(
+                uid=series_uid, description=dcm.SeriesDescription,
+                user_id=current_user.get_id(), device_id=device_id,
+                study_id=study_id, series_datetime=series_datetime)
+            new_series.save()
+            db.session.add(new_series)
+            series_id = new_series.id
+        #TODO remove in production
+        print("series id:", series_id)
+
+        # 3. Image:
+        new_image = Image(
+            uid=image_uid, series_id=series_id, filename=filename,
+            accession_number=accession_number)
+        new_image.save()
+        db.session.add(new_image)
+        #TODO remove in production
+        print("new image id:", new_image.id)
+
+        # Commit all changes to the database
+        db.session.commit()
+
+        # Store file in series/image folder
+        series_folder = Series.query.filter_by(uid=series_uid).first().filesystem_key
+        directory = os.path.join(current_app.config['UPLOADED_PATH'], series_folder)
+        os.makedirs(directory, exist_ok=True)
+
+        return directory
+
+    except Exception as e:
+        raise
+
+
+def locate_image_files(filesystem_key, filename=False):
```

Comment:
same function as before, used to look up image files from the db and then in the file system